src/opening-period/form/OpeningPeriodForm.tsx

```diff
@@ -0,0 +1,322 @@
+  timeSpans: Array<TimeSpanApiFormat>;
+}
+
+type SubmitStatus = 'init' | 'succeeded' | 'error';
+
+type NotificationTexts = {
+  label: string;
+  description: string;
+};
+
+function formatTimeSpansToApiFormat(
+  timeSpans: TimeSpanFormFormat[]
+): TimeSpanApiFormat[] {
+  return timeSpans.map((timeSpan) => {
+    return {
+      description: {
+        fi: timeSpan?.description,
+        sv: null,
+        en: null,
+      },
+      end_time: `${timeSpan.endTime}:00`,
+      start_time: `${timeSpan.startTime}:00`,
+      resource_state: timeSpan.resourceState,
+      weekdays: timeSpan.weekdays.reduce(
+        (acc: Array<number>, currentValue: boolean, currentIndex: number) => {
+          if (currentValue) {
+            acc.push(currentIndex + 1);
+            return acc;
+          }
+          return acc;
+        },
+        []
+      ),
+    };
+  });
+}
+
+const dropMilliseconds = (time: string): string => time.slice(0, -3);
+const getLowestWeekdayNumber = (timeSpan: TimeSpanApiFormat): number =>
+  timeSpan && timeSpan.weekdays ? timeSpan.weekdays.sort()[0] : 0;
+
+function formatApiTimeSpansToFormFormat(
+  apiTimeSpans: TimeSpanApiFormat[]
+): TimeSpanFormFormat[] {
+  return apiTimeSpans
+    .sort((a: TimeSpanApiFormat, b: TimeSpanApiFormat) => {
+      return getLowestWeekdayNumber(a) - getLowestWeekdayNumber(b);
+    })
+    .map((apiTimeSpan) => {
+      const weekdays: boolean[] = Array(7)
+        .fill(false)
+        .map((current, index): boolean =>
+          apiTimeSpan.weekdays
+            ? apiTimeSpan.weekdays.includes(index + 1)
+            : false
+        );
+
+      return {
+        id: apiTimeSpan.id,
+        description: apiTimeSpan.description?.fi || '',
+        startTime: apiTimeSpan.start_time
+          ? dropMilliseconds(apiTimeSpan.start_time)
+          : '',
+        endTime: apiTimeSpan.end_time
+          ? dropMilliseconds(apiTimeSpan.end_time)
+          : '',
+        resourceState: apiTimeSpan.resource_state || ResourceState.OPEN,
+        weekdays: weekdays as FormWeekdays,
+      };
+    });
+}
+
+const formatResourcePeriodOptionsToFormFormat = (
+  resourceStateApiOptions: ResourceStateApiOption[]
+): ResourceStateOption[] =>
+  resourceStateApiOptions.map((optionInApiFormat: ResourceStateApiOption): {
+    value: string;
+    label: string;
+  } => {
+    return {
+      value: optionInApiFormat.value,
+      label: `${
+        typeof optionInApiFormat.display_name === 'string'
+          ? optionInApiFormat.display_name
+          : optionInApiFormat.display_name.fi
+      }`,
+    };
+  });
+
+export default function OpeningPeriodForm({
+  formId,
+  datePeriod,
+  resourceId,
+  datePeriodOptions,
+  submitFn,
+  successTexts,
+  errorTexts,
+}: {
+  formId: string;
+  datePeriod?: DatePeriod;
+  resourceId: number;
+  datePeriodOptions: DatePeriodOptions;
+  submitFn: (datePeriod: DatePeriod) => Promise<DatePeriod>;
+  successTexts: NotificationTexts;
```

Comment:
I wonder would it be more explicit to have separate props for successLabel and successText? Something like submitSuccessNotificationText and submitSuccessNotificationLabel hmm.. Maybe same for errorTexts?

Reply:
Then again Im not sure if it's better to increase the amount of props. One option would be to just rename the successTexts to successTextAndLabel.

Reply:
I'll figure something out

Reply:
Renamed as submitSuccessNotificationText etc.